quickwit-metastore/src/metastore/mod.rs

```diff
@@ -194,6 +194,15 @@ pub struct MetadataSet {
 #[cfg_attr(any(test, feature = "testsuite"), mockall::automock)]
 #[async_trait]
 pub trait Metastore: Send + Sync + 'static {
+    /// Checks if the metastore is available.
+    async fn check(&self) -> anyhow::Result<()>;
+
+    /// Checks if the given index is in this metastore.
+    async fn check_index(&self, index_id: &str) -> anyhow::Result<()> {
```

Comment:
Could we change this method to `index_exists(...) -> anyhow::Result<bool> ` instead and use it for the connectivity checks?

Reply:
Probably, but that's outside the scope of this PR. It would require specialized implementation for all metastore implementations. Also what would `Ok(false)` mean? That the metastore is available but the index isn't?

Reply:
Yeah, `Ok(false)` would mean that the index does not exist.